src/term-spec.js

```diff
@@ -113,36 +126,159 @@ declare export class ObjectAssignmentTarget extends Term {
 
 declare export class AssignmentTargetProperty extends Term { }
 
-declare export class AssignmentTargetPropertyIdentifier extends AssignmentTargetProperty { 
+declare export class AssignmentTargetPropertyIdentifier extends AssignmentTargetProperty {
   binding: AssignmentTargetIdentifier;
   init?: Expression;
 }
 
-declare export class AssignmentTargetPropertyProperty extends AssignmentTargetProperty { 
+declare export class AssignmentTargetPropertyProperty extends AssignmentTargetProperty {
   name: PropertyName;
   binding?: ObjectAssignmentTarget | ArrayAssignmentTarget | AssignmentTargetIdentifier | MemberAssignmentTarget | AssignmentTargetWithDefault;
 }
 
 
 
 // class
+
 declare export class ClassExpression extends Expression {
```

Comment:
I don't love renaming `super` to `heritageClauses`. How about we split the TS stuff out into `ClassExpression` and `TsClassExpression` (with shared attributes in `ClassExpressionBase`)?

Alternatively we could type it as `super: Expression | HeritageClause[]` but in a given AST it will only ever be one or the other so it's nice to keep that clear with two different nodes.

Reply:
I made the split of both `ClassExpression` and `ClassDeclaration`. Once that was done, it also seemed natural to split `ClassElement`, so I did that, too. The nice thing about the result is that it is backwards-compatible with the previous version of classes.

I also added `ClassExpressionBase`, `ClassDeclarationBase` and `ClassElementBase` - they only hold one field each but are perhaps still more preferable than duplicating those fields (and later triplicating for `Flow*` nodes).